addons/crm_mail_client_extension/controllers/main.py

```diff
@@ -6,31 +6,111 @@
 from odoo import http
 from odoo.http import request
 from odoo.tools.misc import formatLang
+from odoo.tools import html2plaintext
+
+from odoo.addons.mail_client_extension.controllers import mail_client_extension
 
 _logger = logging.getLogger(__name__)
 
 
-class MailClientExtensionController(http.Controller):
+class MailClientExtensionController(mail_client_extension.MailClientExtensionController):
 
-    @http.route('/mail_client_extension/log_single_mail_content', type="json", auth="outlook", cors="*")
+    @http.route(route='/mail_client_extension/log_single_mail_content',
+                type="json", auth="outlook", cors="*")
     def log_single_mail_content(self, lead, message, **kw):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
         crm_lead = request.env['crm.lead'].browse(lead)
         crm_lead.message_post(body=message)
 
     @http.route('/mail_client_extension/lead/get_by_partner_id', type="json", auth="outlook", cors="*")
-    def crm_lead_get_by_partner_id(self, partner, limit, offset, **kwargs):
-        partner_leads = request.env['crm.lead'].search([('partner_id', '=', partner)], offset=offset, limit=limit)
+    def crm_lead_get_by_partner_id(self, partner, limit=5, offset=0, **kwargs):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        partner_instance = request.env['res.partner'].browse(partner)
+        return {'leads': self._fetch_partner_leads(partner_instance, limit, offset)}
+
+    @http.route('/mail_client_extension/lead/create_from_partner', type='http', auth='user', methods=['GET'])
+    def crm_lead_redirect_create_form_view(self, partner_id):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        server_action = http.request.env.ref("crm_mail_client_extension.lead_creation_prefilled_action")
+        return werkzeug.utils.redirect('/web#action=%s&model=crm.lead&partner_id=%s' % (server_action.id, int(partner_id)))
+
+    @http.route('/mail_client_extension/lead/create', type='json', auth='outlook', cors="*")
```

Comment:
Wild proposal: rename new routes to something shorted, like
```mail_client/lead/create```

Reply:
Done,

all route names have been changed, old route names are still kept in order to do not break older versions of the plugin.

cc @std-odoo